### PR TITLE
Dev containerservice microsoft.container service 2021 02 01

### PR DESCRIPTION
**Pull request by bcho**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

We want to publish new api version 2021-02-01 for AKS. 

### Changelog

Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by bcho** [comment]
@PhoenixHe-msft could you help review this pr? We want to merge the 0201 api changes to master. Thanks!

**Comment by PhoenixHe-msft** [comment]
@msyyc Python track2 failed. Could you please help to investigate?

**Review by PhoenixHe-NV — APPROVED** [review]

**Comment by bcho** [comment]
hey @msyyc @PhoenixHe-msft any updates for this PR? Should we wait until we fix the python-track2 pipeline failure?

**Review by JunSun17 — COMMENTED** [review]
Looks good to me.

**Review by JunSun17 — APPROVED** [review]
Looks good to me.

**Comment by bcho** [comment]
Hey, I noticed that there are breaking changes in the generated python sdk for 0201 API:

1.  `Components1Umhcm8SchemasManagedclusteridentityPropertiesUserassignedidentitiesAdditionalproperties`

This model is used to call `ManagedClusterIdentityUserAssignedIdentitiesValue` in 20201101, but now, it has been generated to `Components1Umhcm8SchemasManagedclusteridentityPropertiesUserassignedidentitiesAdditionalproperties`

2. breaking changes on the service client:

```
$ az aks show --subscription XXX -g XXX -n XXX
'ServiceClient' object has no attribute '_pipeline'
Traceback (most recent call last):
  File "/usr/local/Cellar/azure-cli/2.13.0/libexec/lib/python3.8/site-packages/knack/cli.py", line 215, in invoke
    cmd_result = self.invocation.execute(args)
  File "/usr/local/Cellar/azure-cli/2.13.0/libexec/lib/python3.8/site-packages/azure/cli/core/commands/__init__.py", line 654, in execute
    raise ex
  File "/usr/local/Cellar/azure-cli/2.13.0/libexec/lib/python3.8/site-packages/azure/cli/core/commands/__init__.py", line 718, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
...
```

I had a draft PR for above changes and we can easily reproduce above issues: 

https://github.com/Azure/azure-cli-extensions/pull/2879


Do you have any idea about what caused this? @PhoenixHe-msft  @JunSun17 Or could you point me someone I should ask for help? Thanks!

**Comment by JunSun17** [comment]
@bcho @msyyc  @PhoenixHe-msft from the log, the breaking change is:

info	[Changelog] **Features**
info	[Changelog]
info	[Changelog]   - Model AgentPool has a new parameter kubelet_disk_type
info	[Changelog]   - Model AgentPool has a new parameter node_public_ip_prefix_id
info	[Changelog]   - Model AgentPool has a new parameter enable_encryption_at_host
info	[Changelog]   - Model Components1Q1Og48SchemasManagedclusterAllof1 has a new parameter fqdn_subdomain
info	[Changelog]   - Model Components1Q1Og48SchemasManagedclusterAllof1 has a new parameter azure_portal_fqdn
info	[Changelog]   - Model ManagedClusterAgentPoolProfileProperties has a new parameter kubelet_disk_type
info	[Changelog]   - Model ManagedClusterAgentPoolProfileProperties has a new parameter node_public_ip_prefix_id
info	[Changelog]   - Model ManagedClusterAgentPoolProfileProperties has a new parameter enable_encryption_at_host
info	[Changelog]   - Model KubeletConfig has a new parameter pod_max_pids
info	[Changelog]   - Model KubeletConfig has a new parameter container_log_max_files
info	[Changelog]   - Model KubeletConfig has a new parameter container_log_max_size_mb
info	[Changelog]   - Model SysctlConfig has a new parameter net_core_rmem_default
info	[Changelog]   - Model SysctlConfig has a new parameter net_core_wmem_default
info	[Changelog]   - Model ManagedClusterAgentPoolProfile has a new parameter kubelet_disk_type
info	[Changelog]   - Model ManagedClusterAgentPoolProfile has a new parameter node_public_ip_prefix_id
info	[Changelog]   - Model ManagedClusterAgentPoolProfile has a new parameter enable_encryption_at_host
info	[Changelog]   - Model ManagedClusterPodIdentityProfile has a new parameter allow_network_plugin_kubenet
info	[Changelog]   - Model ManagedCluster has a new parameter fqdn_subdomain
info	[Changelog]   - Model ManagedCluster has a new parameter azure_portal_fqdn
info	[Changelog]   - Model ManagedClusterPropertiesAutoScalerProfile has a new parameter max_node_provision_time
info	[Changelog]   - Added operation group MaintenanceConfigurationsOperations
info	[Changelog]
info	[Changelog] **Breaking changes**
info	[Changelog]
**_info	[Changelog]   - Model SysctlConfig no longer has parameter net_ipv4_tcp_rmem
info	[Changelog]   - Model SysctlConfig no longer has parameter net_ipv4_tcp_wmem_**

**Comment by JunSun17** [comment]
Checked a bit further, and looks the difference on SysctlConfig is introduced in 2020-11-01 to 2020-12-01 API change. Do not know why it is not being caught at that time. Looks the current 2021-02-01 client generation include changes coming from both 2020-12-01 and 2021-02-01. @bcho @PhoenixHe-msft

**Comment by alexeldeib** [comment]
@PhoenixHe-msft track2 failure is because python SDK is comparing 2021-02-01 API against 2020-11-01 and not against 2020-12-01. The diff is the same breaking change we already covered in https://github.com/Azure/azure-rest-api-specs/pull/12163. 

Here is the Python SDK diff showing it's still using the 2020-11-01 API: https://github.com/AzureSDKAutomation/azure-sdk-for-python/pull/4129/files#diff-22cab198d39348ef25897451c1708725d3d368f16424351a4b49e7d566ffe527L49

**Comment by PhoenixHe-msft** [comment]
Yes I know these pending breaking changes that are not caused by this PR. We are going to introduce breaking change detection in same PR. For this PR, the breaking change is not an issue. Now you need to wait for ARM review.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2021-02-01/managedClusters.json [hunk 1]
```diff
@@ -0,0 +1,4120 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "ContainerServiceClient",
+    "description": "The Container Service Client.",
+    "version": "2021-02-01"
```
Comment (RamyasreeChakka) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2021-02-01/managedClusters.json:6]: @bcho - For quicker API reviews, we recommend to fill the change log informing the changes made in this PR.

Comment (bcho) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2021-02-01/managedClusters.json:6]: Hi @RamyasreeChakka , I had went through and updated the check list in the PR description. PTAL!

**Comment by JunSun17** [comment]
@PhoenixHe-msft ARM has approved this PR. Can you help to merge? Thanks!

**Pull request merged**